agency/test/services/credential.test.ts

```diff
@@ -1,8 +1,267 @@
+    const params = {
+      profile: profileWithSchema,
+      headers: context.params.headers
+    } as unknown as IssuerServiceParams;
+
+    jest.spyOn(sleep, 'deferServiceOnce')
+      .mockReset()
+      .mockImplementationOnce((id) => mockSuccess(params, id))
+      .mockImplementationOnce((id) => mockSuccess(params, id));
+
+    const credentials = [
+      { ...credential, reservation_number: '000002' },
+      { ...credential, reservation_number: '000003' }
+    ];
+    await credentialService.create(credentials, params);
+    expect(params.credentials.results).toBeInstanceOf(Array);
+    expect(params.credentials.results).toEqual([
+      { cred_ex_id: 'credential-exchange-id', success: true, error: undefined },
+      { cred_ex_id: 'credential-exchange-id', success: true, error: undefined }
+    ]);
+  });
+
+  it('should return a valid response even if a credential is not succesffully issued', async () => {
+    const credentialService = app.service('issuer/credential');
+    const profileWithSchema = await app.service('issuer/model').get(profile._id);
+    const params = {
+      profile: profileWithSchema,
+      headers: context.params.headers
+    } as unknown as IssuerServiceParams;
+
+    jest.spyOn(sleep, 'deferServiceOnce')
+      .mockReset()
+      .mockImplementationOnce((id) => mockSuccess(params, id))
+      .mockImplementationOnce((id) => mockError(params, id));
+
+    const credentials = [
+      { ...credential, reservation_number: '000003' },
+      { ...credential, reservation_number: '000004' }
+    ];
+    await credentialService.create(credentials, params);
+    expect(params.credentials.results).toBeInstanceOf(Array);
+    expect(params.credentials.results).toEqual([
+      { cred_ex_id: 'credential-exchange-id', success: true, error: undefined },
+      { cred_ex_id: 'credential-exchange-id', success: false, error: undefined }
+    ]);
+  });
+
+  it('should return a valid response even if a credential is not succesffully issued', async () => {
+    const credentialService = app.service('issuer/credential');
+    const profileWithSchema = await app.service('issuer/model').get(profile._id);
+    const params = {
+      profile: profileWithSchema,
+      headers: context.params.headers
+    } as unknown as IssuerServiceParams;
+
+    jest.spyOn(sleep, 'deferServiceOnce')
+      .mockReset()
+      .mockImplementationOnce((id) => mockError(params, id))
+      .mockImplementationOnce((id) => mockError(params, id));
+
+    const credentials = [
+      { ...credential, reservation_number: '000005' },
+      { ...credential, reservation_number: '000006' }
+    ];
+    await credentialService.create(credentials, params);
+    expect(params.credentials.results).toBeInstanceOf(Array);
+    expect(params.credentials.results).toEqual([
+      { cred_ex_id: 'credential-exchange-id', success: false, error: undefined },
+      { cred_ex_id: 'credential-exchange-id', success: false, error: undefined }
+    ]);
+  });
```

Comment:
These seem like they may be testing the same scenario twice. If we want to keep them, they should be named differently so we can tell them apart easier.